core/src/main/scala/chisel3/internal/package.scala

```diff
@@ -143,8 +143,11 @@ package object internal {
         color
       case _ => return
     }
-    if (!Builder.layerStack.exists(_ == destLayer))
-      Builder.error(errorMessage)
+    val enabledLayers = Builder.enabledLayers ++= Builder.layerStack.headOption
```

Comment:
I don't fully grok this mutation, shouldn't the enabling of layers happen separately from the checking of them?

Reply:
This is actually wrong. It should just be `++`. No mutation. Will fix...

Reply:
Fixed in https://github.com/chipsalliance/chisel/pull/3857/commits/023776502bd2207a37222eeca60b598bb1cffc22.